pyplis/imagelists.py

```diff
@@ -1608,21 +1608,15 @@ def _iter_num(self, start_idx, stop_idx):
 
     def _first_file(self):
         """Get first file path of image list."""
-        try:
-            return self.files[0]
-        except IndexError:
-            logger.warning("Filelist empty...")
-        except BaseException:
-            raise
+        if not bool(self.files):
+            raise IndexError('ImgList is empty...')
+        return self.files[0]
 
     def _last_file(self):
         """Get last file path of image list."""
-        try:
-            return self.files[self.nof - 1]
-        except IndexError:
-            logger.warning("Filelist empty...")
-        except:
-            raise
+        if not bool(self.files):
```

Comment:
are you sure you actually need that book? Wouldn't that be cast implicitly?

Reply:
It would :) But this way it will throw an easier to understand exception (at least that's what I hope) ;)